base/int.jl

```diff
@@ -131,13 +131,40 @@ fld(x::Unsigned, y::Signed) = div(x, y) - (signbit(y) & (rem(x, y) != 0))
 
 """
     mod(x, y)
+    rem(x, y, RoundDown)
 
-Modulus after flooring division, returning in the range ``[0,y)``, if `y` is positive, or
-``(y,0]`` if `y` is negative.
+The reduction of `x` modulo `y`, or equivalently, the remainder of `x` after floored
+division by `y`. Returns a value with the same sign as `y` whose magnitude is less than
+`abs(y)` (except possibly in the case of floating point values, see note below).
 
+This has the property that
 ```julia
 x == fld(x,y)*y + mod(x,y)
 ```
+if computed without intermediate rounding.
+
+!!! note
+
+    When used with floating point values, the exact result may not be representable by the
+    type, and so rounding error may occur. In particular, if the exact result is very
```

Comment:
the return type, which will be floating point if either `x` or `y` are floating point (or any numeric value which gets promoted to floating point). Or any wrapper around a floating point value (e.g. dual or unitful numbers).

I intentionally changed it from being too specific as I think it was getting too bogged down in minutae, rather than getting the point across.